shared/wallets/wallet/container.js

```diff
@@ -26,8 +26,15 @@ const mergeProps = (stateProps, dispatchProps) => {
 
   const [completed, pending] = partition(payments, {status: 'completed'})
 
-  sections.push({data: pending, title: 'Pending'})
-  sections.push({data: completed, title: 'History'})
+  if (pending.length > 0) {
+    sections.push({data: pending, title: 'Pending'})
+  }
+
+  if (completed.length === 0) {
```

Comment:
Best place to do these length checks seemed to be in the container, but let me know if they should go somewhere else